recipes/tz/all/conanfile.py

```diff
@@ -50,24 +57,63 @@ def _patch_sources(self):
         replace_in_file(self, os.path.join(self.source_folder, "Makefile"), "AWK=		awk", f"AWK={awk_path}")
 
     def build(self):
-        self._patch_sources()
-        autotools = Autotools(self)
-        autotools.make(args=["-C", self.source_folder.replace("\\", "/")])
+        if self.options.with_binary_db:
+            self._patch_sources()
+            autotools = Autotools(self)
+            ldlibs = ""
+            if self.settings.os == "Macos":
+                ldlibs = "-lintl"
```

Comment:
```suggestion
            if self.settings.os == "Macos":
                ldlibs = "-lintl"
```

It does not work locally:

```
% conan create all --version=2023c -o tz/2023c:with_binary_db=True 
======== Exporting recipe to the cache ========
tz/2023c: Exporting package recipe: /Users/uilian/Development/conan/conan-center-index/recipes/tz/all/conanfile.py
tz/2023c: exports: File 'conandata.yml' found. Exporting it...
tz/2023c: Copied 1 '.py' file: conanfile.py
tz/2023c: Copied 1 '.yml' file: conandata.yml
tz/2023c: Exported to cache folder: /Users/uilian/.conan2/p/tzfd671b2e5f670/e
tz/2023c: Exported: tz/2023c#267eec9fdf74ff1c19025e1209cc5e09 (2024-01-17 12:14:16 UTC)

======== Input profiles ========
Profile host:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=14
os=Macos
[options]
tz/2023c:with_binary_db=True
[conf]
tools.build:compiler_executables={'fortran': '/opt/homebrew/bin/gfortran-11'}

Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.cppstd=17
compiler.libcxx=libc++
compiler.version=14
os=Macos
[conf]
tools.build:compiler_executables={'fortran': '/opt/homebrew/bin/gfortran-11'}


======== Computing dependency graph ========
Graph root
    cli
Requirements
    tz/2023c#267eec9fdf74ff1c19025e1209cc5e09 - Cache
Build requirements
    mawk/1.3.4-20230404#62d52d57a28003122376f143fa6e1857 - Cache

======== Computing necessary packages ========
tz/2023c: Forced build from source
Requirements
    tz/2023c#267eec9fdf74ff1c19025e1209cc5e09:1b44a2747e2ceeacc731b1c2eaf4782ab3fec17f - Build
Build requirements
    mawk/1.3.4-20230404#62d52d57a28003122376f143fa6e1857:617cae191537b47386c088e07b1822d8606b7e67#c1b93a18140ff1988cebd889aed0c36b - Cache

======== Installing packages ========
mawk/1.3.4-20230404: Already installed! (1 of 2)

-------- Installing package tz/2023c (2 of 2) --------
tz/2023c: Building from source
tz/2023c: Package tz/2023c:1b44a2747e2ceeacc731b1c2eaf4782ab3fec17f
tz/2023c: Copying sources to build folder
tz/2023c: Building your package in /Users/uilian/.conan2/p/b/tz0605e684855d8/b
tz/2023c: Calling generate()
tz/2023c: Generators folder: /Users/uilian/.conan2/p/b/tz0605e684855d8/b/build-release/conan
tz/2023c: Generating aggregated env files
tz/2023c: Generated aggregated env files: ['conanbuild.sh', 'conanrun.sh']
tz/2023c: Calling build()
tz/2023c: RUN: make -C /Users/uilian/.conan2/p/b/tz0605e684855d8/b/src LDLIBS=-lintl -j10
{ (type git) >/dev/null 2>&1 && \
		  V=`git describe --match '[0-9][0-9][0-9][0-9][a-z]*' \
				--abbrev=7 --dirty` || \
		  if test 'unknown' = unknown && V=`cat version`; then \
		    case $V in *-dirty);; *) V=$V-dirty;; esac; \
		  else \
		    V='unknown'; \
		  fi; } && \
		printf '%s\n' "$V" >version.out
cc -DTZDIR='"/usr/share/zoneinfo"'   -DNDEBUG  -c -o localtime.o localtime.c
cc -DTZDIR='"/usr/share/zoneinfo"'   -DNDEBUG  -c -o asctime.o asctime.c
cc -DTZDIR='"/usr/share/zoneinfo"'   -DNDEBUG  -c -o strftime.o strftime.c
cc -DTZDIR='"/usr/share/zoneinfo"'   -DNDEBUG  -c -o difftime.o difftime.c
/Users/uilian/.conan2/p/mawkcd72ee49b16ee/p/bin/mawk -v EXPIRES_LINE=0 \
		  -f leapseconds.awk leap-seconds.list >leapseconds.out
/Users/uilian/.conan2/p/mawkcd72ee49b16ee/p/bin/mawk \
		  -v DATAFORM=`expr main.zi : '\(.*\).zi'` \
		  -v PACKRATDATA='' \
		  -v PACKRATLIST='' \
		  -f ziguard.awk \
		  africa antarctica asia australasia europe northamerica southamerica etcetera factory backward  >main.zi.out
/Users/uilian/.conan2/p/mawkcd72ee49b16ee/p/bin/mawk \
		  -v DATAFORM=`expr vanguard.zi : '\(.*\).zi'` \
		  -v PACKRATDATA='' \
		  -v PACKRATLIST='' \
		  -f ziguard.awk \
		  africa antarctica asia australasia europe northamerica southamerica etcetera factory backward  >vanguard.zi.out
/Users/uilian/.conan2/p/mawkcd72ee49b16ee/p/bin/mawk \
		  -v DATAFORM=`expr rearguard.zi : '\(.*\).zi'` \
		  -v PACKRATDATA='' \
		  -v PACKRATLIST='' \
		  -f ziguard.awk \
		  africa antarctica asia australasia europe northamerica southamerica etcetera factory backward  >rearguard.zi.out
fatal: not a git repository (or any of the parent directories): .git
mv leapseconds.out leapseconds
cat: version: No such file or directory
mv version.out version
VERSION=`cat version` && sed \
			-e 's|#!/bin/bash|#!/bin/bash|g' \
			-e 's|AWK=[^}]*|AWK='\''/Users/uilian/.conan2/p/mawkcd72ee49b16ee/p/bin/mawk'\''|g' \
			-e 's|\(PKGVERSION\)=.*|\1='\''(tzcode) '\''|' \
			-e 's|\(REPORT_BUGS_TO\)=.*|\1=tz@iana.org|' \
			-e 's|TZDIR=[^}]*|TZDIR=/usr/share/zoneinfo|' \
			-e 's|\(TZVERSION\)=.*|\1='"$VERSION"'|' \
			<tzselect.ksh >tzselect.out
VERSION=`cat version` && printf '%s\n' \
		  'static char const PKGVERSION[]="(tzcode) ";' \
		  "static char const TZVERSION[]=\"$VERSION\";" \
		  'static char const REPORT_BUGS_TO[]="tz@iana.org";' \
		  >version.h.out
mv version.h.out version.h
chmod +x tzselect.out
mv main.zi.out main.zi
cc -DTZDIR='"/usr/share/zoneinfo"'   -DNDEBUG  -c -o zic.o zic.c
mv tzselect.out tzselect
cc -DTZDIR='"/usr/share/zoneinfo"'   -DNDEBUG  -c -o zdump.o zdump.c
version=`sed 1q version` && \
		  LC_ALL=C /Users/uilian/.conan2/p/mawkcd72ee49b16ee/p/bin/mawk \
		    -v dataform='main' \
		    -v deps='ziguard.awk africa antarctica asia australasia europe northamerica southamerica etcetera factory backward   zishrink.awk' \
		    -v redo='posix_right' \
		    -v version="$version" \
		    -f zishrink.awk \
		    main.zi >tzdata.zi.out
mv rearguard.zi.out rearguard.zi
mv vanguard.zi.out vanguard.zi
mv tzdata.zi.out tzdata.zi
cc -DTZDIR='"/usr/share/zoneinfo"' -o zdump   zdump.o localtime.o asctime.o strftime.o -lintl
rm -f libtz.a
ar -rc libtz.a localtime.o asctime.o difftime.o strftime.o
: libtz.a
cc -DTZDIR='"/usr/share/zoneinfo"' -o zic   zic.o -lintl
ld: library 'intl' not found
ld: library 'intl' not found
clang: error: clang: error: linker command failed with exit code 1 (use -v to see invocation)linker command failed with exit code 1 (use -v to see invocation)

make: *** [zic] Error 1
make: *** Waiting for unfinished jobs....
make: *** [zdump] Error 1

tz/2023c: ERROR: 
Package '1b44a2747e2ceeacc731b1c2eaf4782ab3fec17f' build failed
tz/2023c: WARN: Build folder /Users/uilian/.conan2/p/b/tz0605e684855d8/b/build-release
*********************************************************
Recipe 'tz/2023c' cannot build its binary
It is possible that this recipe is not Conan 2.0 ready
If the recipe comes from ConanCenter, report it at https://github.com/conan-io/conan-center-index/issues
If it is your recipe, check if it is updated to 2.0
*********************************************************

ERROR: tz/2023c: Error in build() method, line 66
	autotools.make(args=["-C", self.source_folder.replace("\\", "/"), f"LDLIBS={ldlibs}"])
	ConanException: Error 2 while executing
```

Reply:
Hmm. I suspect the requirement for this may be more narrow than `os == Macos` in that case. When I tested this it was required to get it building locally on my x86_64 mac, and I added it at the guidance of [the Makefile](https://github.com/eggert/tz/blob/8459570715214271fd5bab92ec4d6643901c02e0/Makefile#L209):

> # Non-default libraries needed to link.
> # On some hosts, this should have -lintl unless CFLAGS has -DHAVE_GETTEXT=0.
> LDLIBS=

I can remove it and see how we go

Reply:
I've reverted this change. I'll see how this goes with CI when https://github.com/conan-io/hooks/pull/525 is approved